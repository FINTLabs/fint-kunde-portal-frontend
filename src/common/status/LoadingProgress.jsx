import PropTypes from "prop-types";
import React from "react";
import { CircularProgress } from "@mui/material";
import { styled } from '@mui/material/styles';

const CustomCircularProgress = styled(CircularProgress)(({ theme }) => ({
  position: "absolute",
  top: "30%",
  left: "50%",
  marginRight: "-50%"
}));


class LoadingProgress extends React.Component {
  render() {
    const { color, size } = this.props;

    return (
      <CustomCircularProgress
        color={color}
        size={size}
      />
    );
  }
}

LoadingProgress.propTypes = {
  classes: PropTypes.object,
  color: PropTypes.string,
  size: PropTypes.number
};

LoadingProgress.defaultProps = {
  color: "primary",
  size: 125
};

export default (LoadingProgress);
