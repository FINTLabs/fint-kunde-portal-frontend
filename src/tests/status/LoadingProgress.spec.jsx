import React from "react";
import { create } from "react-test-renderer";
import LoadingProgress from "../../common/status/LoadingProgress";
describe("Feature component", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <LoadingProgress
        color="primary"
        size={10}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});