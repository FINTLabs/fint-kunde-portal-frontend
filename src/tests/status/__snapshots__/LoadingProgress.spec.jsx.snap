// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component it matches the snapshot 1`] = `
<div
  className="MuiCircularProgress-root LoadingProgress-progress-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 10,
      "width": 10,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
